### ISSUE TITLE
Migrate website to astro

### ISSUE THREAD

**Issue by timotheeguerin** [microsoft/typespec#2564]
After some quick investigation migrating to astro would both improve build performance and runtime performance.

Astro main selling point is everything is rendered on the server(unless go through extra step to have dynamic content)
This means it result in a completely static website without the need for javascript. Somethings like light/dark theme switch still use js but the core of the website is available.

Pros:
- Build perf 2:30min(docusaurus) -> 50s(astro)  - tested on m1 pro
  -  EDIT: build perf is actually very bad when the sidebar is included
- Use tm grammar for syntax highlighting (no more prismjs with half baked syntax highlighting)

Issues:
- [ ] Doesn't support multiple sidebar or versioning which make migration today impossible without a lot of manual work. There are some active issues for those problem and it seems to be getting traction.
- [x] Auto generated sidebar doesn't have good names for folders. This could be mitigated by having our own custom resolver

```[tasklist]
- [x] Initial side by side migration https://github.com/microsoft/typespec/pull/4587
- [x] Migrate more react component to astro https://github.com/microsoft/typespec/pull/4710
- [x] Add blog https://github.com/microsoft/typespec/pull/4797
- [x] Header responsive https://github.com/microsoft/typespec/pull/4800
- [x] astro prettier plugin
- [x] Switch which version gets published
- [x] Remove docusaurus
- [x] Move docs/ folder? https://github.com/microsoft/typespec/pull/4928
```

**Comment by rishi-raj-jain** [comment]
I'd love to be part of this effort.

**Comment by rishi-raj-jain** [comment]
> Doesn't support multi versions

Why's that the case? I believe you can use starlight.astro.build and just create folders for versioning, ain't it?

**Comment by timotheeguerin** [comment]
Yeah we can just need to then do the selection ourself. The first item was the biggest deal breaker the version might just need more or less work from us
